runtime/array.c

```diff
@@ -233,7 +233,7 @@ CAMLprim value caml_array_make(value len, value init)
         CAML_EV_COUNTER (EV_C_FORCE_MINOR_MAKE_VECT, 1);
         caml_minor_collection ();
       }
-      CAMLassert(!(Is_block(init) && Is_young(init)));
+      CAMLassert(!Is_block(init) || !Is_young(init));
```

Comment:
This is a gratuitous change. Didn't you want to split such asserts in two?

Reply:
No, I'm splitting conjunctions, but splitting disjunctions is nastier… I'm just rewriting `¬(A ∧ B)` to `¬A ∨ ¬B` to make that clearer, and it's easier to grep for `CAMLassert.*\&\&`.

Reply:
Fair enough.

Reply:
I find the previous version easier to read: `Is_young(foo)` can only be used when we know that `Is_block(foo)` holds, and it is easier to check that this precondition is respected in the previous version.

Reply:
Reverted.